PyRoute/Star.py

```diff
@@ -749,17 +832,50 @@ def _int_to_ehex(self, value):
     def split_stellar_data(self):
         star_parts = self.stars.split()
         stars = []
-
-        if len(star_parts) == 1:
+        types = ('V', 'IV', 'Ia', 'Ib', 'II', 'III', 'VI')
+        class_set = (
+            'O6', 'O7', 'O8', 'O9',
+            'B0', 'B1', 'B2', 'B3', 'B4', 'B5', 'B6', 'B7', 'B8', 'B9',
+            'A0', 'A1', 'A2', 'A3', 'A4', 'A5', 'A6', 'A7', 'A8', 'A9',
+            'F0', 'F1', 'F2', 'F3', 'F4', 'F5', 'F6', 'F7', 'F8', 'F9',
+            'G0', 'G1', 'G2', 'G3', 'G4', 'G5', 'G6', 'G7', 'G8', 'G9',
+            'K0', 'K1', 'K2', 'K3', 'K4', 'K5', 'K6', 'K7', 'K8', 'K9',
+            'M0', 'M1', 'M2', 'M3', 'M4', 'M5', 'M6', 'M7', 'M8', 'M9',
+        )
+        star_parts = [part for part in star_parts if part in class_set or part in Star.oddball_stars or part in types]
+
+        if 1 == len(star_parts) and star_parts[0] in class_set:
+            self.logger.warning("Code " + star_parts[0] + " lacks explicit star class, inserting V")
+            # Most common star class is V, so using that here
+            star_parts.append('V')
+        if 1 == len(star_parts) and star_parts[0] in types:
+            star_parts = []
+
+        if len(star_parts) < 2:
             stars = star_parts
         else:
             for (prev, current) in zip([None] + star_parts[:-1], star_parts):
-                if prev in [None, 'V', 'IV', 'Ia', 'Ib', 'II', 'III']:
+                if prev in [None, 'V', 'IV', 'Ia', 'Ib', 'II', 'III', 'VI']:
```

Comment:
This is `types` defined above.
```suggestion
                if prev is None or prev in types:
```

Reply:
Fixed.  Thanks for spotting this.